### PR TITLE
feat: tutorials-integration

### PR DESCRIPTION
**Pull request by gratcliff**
## 🧰 What's being changed?
- Integrating `@readme/ui` into our `api-explorer` subpackage
- Implementing `tutorials` `tile` and `modal` views for `tutorial integration`. This data comes via the `docs.tutorials` collection.
- Adding an additional `@readme/variable` provider to populate `TutorialModal` `readonly view` variables with user data.

## 🧪 Testing

Provide as much information as you can on how to test what you've done.

### CODE REVIEW

**Comment by gratcliff** [comment]
Todo:
1. Reduce bundle size via optimizing transpiled `@readme/ui` output
2. Include exported `context` to populate our `@readme/variables` in the `syntax-highlighter`
3. Potentially add advertising callouts.

**Comment by runnabro** [comment]
@gratcliff Some padding + overflow

Screen Shot 2020-10-19 at 4 37 34 PM

**Comment by gratcliff** [comment]
> @gratcliff Some padding + overflow
> 
> Screen Shot 2020-10-19 at 4 37 34 PM

Nice!! Looks great. Do they scroll horizontally?

**Comment by runnabro** [comment]
@gratcliff yeah, they do (tested on chrome, ff, safari)

**Review by erunion — CHANGES_REQUESTED** [review]
I added an experiment to the demo app for making tutorials visible but when clicking on one, a `codeOptions` error is thrown from the `TutorialModal` component. Not sure if this'll happen when the explorer is running in a non-demo app, if it's a problem with the fake tutorial data or what.

Screen Shot 2020-10-21 at 11 54 59 AM

**Comment by gratcliff** [comment]
> I added an experiment to the demo app for making tutorials visible but when clicking on one, a `codeOptions` error is thrown from the `TutorialModal` component. Not sure if this'll happen when the explorer is running in a non-demo app, if it's a problem with the fake tutorial data or what.
> 
> Screen Shot 2020-10-21 at 11 54 59 AM

@erunion looks like you'll need the full `tutorial` example object to see the `modal`, I may have led you astray in our slack chat!

File: packages/api-explorer/src/Doc.jsx [hunk 1]
```diff
@@ -368,6 +381,13 @@ class Doc extends React.Component {
                 </div>
               )}
             </header>
+            <div className="TutorialTile-Container">
```
[the remark this answers is not in the document]
Comment (gratcliff) [reply packages/api-explorer/src/Doc.jsx]: Fixed this padding issue!

**Comment by gratcliff** [comment]
@erunion fixed the `tutorial modal` demo issue by updating to a full schema reference. Also added in additional check in the demo to wipe the `doc.tutorials` array so that our state would update properly.

**Review by erunion — APPROVED** [review]

**Pull request merged**